### ISSUE TITLE
Rule extension proposal: id-blacklist option to ignore property names

### ISSUE THREAD

**Issue by ftorrado** [eslint/eslint#7148]
**What version of ESLint are you using?**

^2.11.1

**What problem do you want to solve?**

When using the "id-blacklist" rule, object properties are always included. There should be a way similar to what happens on the "id-length" rule where there is the option to ignore property names. Converting identifiers for existing data models is not an option in many cases.

---

EDIT:

**What rule do you want to change?**

id-blacklist

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option: `{"properties": never}` (`{"properties": always }` is the default)

**Please provide some example code that this change will affect:**

``` js
$list.kendoDropDownList({
  dataTextField: 'name',
            dataSource: {
                    data: listData
            }
        });
```

**What does the rule currently do for this code?**

The "data" id is blacklisted (`"id-blacklist": ["data", ...]`) and eslint shows an warning/error for the "data" property. I want 

**What will the rule do after it's changed?**

No warning will be shown on eslint, it ignores the names of object properties.

**Comment by vitorbal** [comment]
hey @ftorrado, thanks for reaching out!

Could you do us a big favor and fill in the rest of the [template for proposing a rule change](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)?

That will help the team understand what you are proposing and allow us to give quicker feedback on your proposal. Thanks a lot!

**Comment by eslintbot** [comment]
Hi @ftorrado, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by vitorbal** [comment]
@ftorrado thanks for updating your original issue description!

The `id-blacklist` rule already excludes property names that are not assignments, like: `foo.data`, so I think an option named `properties` would be misleading. Maybe something like `propertyAssignments` would be better?

Personally I'm not in favor of this enhancement as I think the `id-blacklist` would lose a lot of its value if the check for all object property assignments is disabled..
I would suggest an inline comment disabling the rule for those specific cases where you need to create an object that will be consumed by a vendor API you have no control of. That way the rule would still be enforced in all object property assignments that don't deal with vendor APIs.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
